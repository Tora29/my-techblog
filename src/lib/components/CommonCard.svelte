<script lang="ts">
	// コンポーネントの変数
	export let link: string = '/';
	export let cardSide: 'card-side' | '' = 'card-side';
	export let color: 'bg-primary' | 'bg-secondary' = 'bg-primary';
	export let customStyle: string = ''
	export let imgSrc: string = '';
	export let imgAlt: string = '';
	export let cardTitle: string = 'card-title';
	export let cardText: string = 'card-text';
</script>

<a href={link} class="card {cardSide} {color}">
	<figure>
		<img src={imgSrc} alt={imgAlt} class="{customStyle}" />
	</figure>
	<div class="card-body">
		<h2 class="card-title">{cardTitle}</h2>
		<p>{cardText}</p>
	</div>
</a>
