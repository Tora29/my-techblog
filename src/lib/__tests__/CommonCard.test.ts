import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import '@testing-library/jest-dom';
import CommonCard from '$lib/components/CommonCard.svelte';

describe('Card Component', () => {
	test('カードをデフォルトプロパティでレンダリング', () => {
		const { getByRole, getByText, container } = render(CommonCard);

		// リンクのデフォルト値を確認
		const link = getByRole('link');
		expect(link).toHaveAttribute('href', '/');

		// デフォルトのタイトルとテキストを確認
		expect(getByText('card-title')).toBeInTheDocument();
		expect(getByText('card-text')).toBeInTheDocument();

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});

	test('カードをカスタムプロパティでレンダリング', () => {
		const props = {
			link: '/test-link',
			cardSide: '',
			color: 'bg-secondary',
			customStyle: 'h-40 w-100',
			imgSrc: 'https://test.com/image.jpg',
			imgAlt: 'Test Image',
			cardTitle: 'Test Title',
			cardText: 'Test text content.'
		} as const;

		const { getByRole, getByAltText, getByText, container } = render(CommonCard, { props });

		// カスタムリンクを確認
		const link = getByRole('link');
		expect(link).toHaveAttribute('href', props.link);
		expect(link).toHaveClass(props.color);

		// カスタム画像を確認
		const img = getByAltText(props.imgAlt);
		expect(img).toHaveAttribute('src', props.imgSrc);
		expect(img).toHaveClass(props.customStyle);

		// カスタムタイトルとテキストを確認
		expect(getByText(props.cardTitle)).toBeInTheDocument();
		expect(getByText(props.cardText)).toBeInTheDocument();

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});
});
