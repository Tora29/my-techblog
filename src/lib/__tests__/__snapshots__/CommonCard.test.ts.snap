// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Component > imgSrcが空の場合、画像がレンダリングされない 1`] = `
<div>
  <a
    class="card card-side bg-primary h-40 w-100"
    href="/"
  >
    <figure>
      <img
        alt=""
        class="h-40 w-72"
      />
    </figure>
     
    <div
      class="card-body"
    >
      <h2
        class="card-title"
      >
        card-title
      </h2>
       
      <p>
        card-text
      </p>
    </div>
  </a>
</div>
`;

exports[`Card Component > カードをカスタムプロパティでレンダリング 1`] = `
<div>
  <a
    class="card card-side bg-secondary h-50 w-200"
    href="/test-link"
  >
    <figure>
      <img
        alt="Test Image"
        class="h-50 w-150"
        src="https://test.com/image.jpg"
      />
    </figure>
     
    <div
      class="card-body"
    >
      <h2
        class="card-title"
      >
        Test Title
      </h2>
       
      <p>
        Test text content.
      </p>
    </div>
  </a>
</div>
`;

exports[`Card Component > カードをデフォルトプロパティでレンダリング 1`] = `
<div>
  <a
    class="card card-side bg-primary h-40 w-100"
    href="/"
  >
    <figure>
      <img
        alt=""
        class="h-40 w-72"
      />
    </figure>
     
    <div
      class="card-body"
    >
      <h2
        class="card-title"
      >
        card-title
      </h2>
       
      <p>
        card-text
      </p>
    </div>
  </a>
</div>
`;
